lib/features/sks-menu/presentation/sks_menu_screen.dart

```diff
@@ -30,17 +32,27 @@ class SksMenuView extends ConsumerWidget {
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     final asyncSksMenuData = ref.watch(getSksMenuDataProvider);
+    final isLastMenuButtonClicked = useState(false);
+
 
     return asyncSksMenuData.when(
-      data: (sksMenuData) => _SksMenuView(
-        asyncSksMenuData.value ??
-            SksMenuResponse(
-              isMenuOnline: false,
-              lastUpdate: DateTime.now(),
-              meals: List.empty(),
-            ),
-        appBarPopTitle,
-      ),
+      data: (sksMenuData) {
+        if (!sksMenuData.isMenuOnline && !isLastMenuButtonClicked.value) {
+          return _SKSMenuLottieAnimation(
+            onShowLastMenuTap: () {
+              isLastMenuButtonClicked.value = true;
+            },
+          );
+        }
+        if (sksMenuData.isMenuOnline && isLastMenuButtonClicked.value) {
+          isLastMenuButtonClicked.value = false;
+        }
```

Comment:
so we probably shouldn't change state during build method itself. For this we place it inside `useEffect` hook for example? Or just adjust the logic for this to be not necessary.

Reply:
So if you wanna change this, then this useEffect hook could probably be a better idea:
```dart
final asyncSksMenuData = ref.watch(getSksMenuDataProvider);
    final isLastMenuButtonClicked = useState(false);
    useEffect(() {
      if ((asyncSksMenuData.value?.isMenuOnline ?? false) &&
          isLastMenuButtonClicked.value) {
        isLastMenuButtonClicked.value = false;
      }
      return null;
    }, [
      asyncSksMenuData.value?.isMenuOnline,
      isLastMenuButtonClicked.value,
    ]);
```
But it's long and nasty.

But if you look down to you other widget, you don't show the animation if the menu is online anyway - so imo you can skip it. I know you wanted to make sure that we cover the edge case that menu is offline, someone clicked, it then went offline and online again and for us this would mean it's clicked - but imo that's not a problem. Apart from this case, this if statement didn't change anything anyway. So imo you can just skip it.